tests/test_examples.py

```diff
@@ -34,39 +36,72 @@ def get_file_paths():
+    """Return the string stripped to the final ']'
+
+    Running Pytest from a Python Console produces a trailing ANSI '\x1b[0m' code whereas
+    the CLI Pytest does not.
+    """
+    trailing_token = str_to_strip.split(']')[-1]
+    return str_to_strip[:-len(trailing_token)]
+
+
+def _compile_message_symbols_per_file(pylint_output_list: List[Dict]) -> Dict[str, Set[str]]:
+    """Return a dict mapping file name to the set of Pylint message symbols it raises.
+
+    The file name is the basename, not the full or relative path
+    (this is to avoid being unable to access a value because of \\ and / paths)
     """
-    # The following are captured when this function is created.
-    def new_test_func():
-        found_pylint_message = False
-        output = subprocess.run(
-            ['pylint', '--reports=n',
-             '--rcfile=python_ta/.pylintrc',
-             test_file],
-            stderr=subprocess.STDOUT,
-            stdout=subprocess.PIPE)
-        for line in output.stdout.decode('utf-8').split('\n'):
-            if checker_name in line:
-                found_pylint_message = True
-                break
-        if not found_pylint_message:
-            print('Failed: ' + test_file)  # test doesn't say which file
-        assert found_pylint_message
-    return new_test_func
+    symbols_by_file = {}
+
+    for pylint_message_data in pylint_output_list:
+        file_name = os.path.basename(pylint_message_data['path'])
+        symbol = pylint_message_data['symbol']
+
+        if file_name not in symbols_by_file:
+            symbols_by_file[file_name] = set()
+
+        symbols_by_file[file_name].add(symbol)
+
+    return symbols_by_file
+
+
+@pytest.fixture(scope='session', autouse=True)
+def symbols_by_file() -> Dict[str, Set[str]]:
+    """Run pylint on all the example files return the map of file name to the
+    set of Pylint messages it raises."""
+    pylint_list_output = _get_full_pylint_json_output()
+    return _compile_message_symbols_per_file(pylint_list_output)
 
 
 @pytest.mark.parametrize("test_file", get_file_paths())
-def test_examples_files(test_file):
+def test_examples_files(test_file: str, symbols_by_file: Dict[str, Set[str]]):
     """Creates all the new unit tests dynamically from the testing directory."""
     base_name = os.path.basename(test_file)
     if not re.match(_EXAMPLE_PREFIX_REGEX, base_name[:5]):
         return
     if not base_name.lower().endswith('.py'):
         assert False
     checker_name = base_name[6:-3].replace('_', '-')  # Take off prefix and file extension.
-    test_function = create_checker(test_file, checker_name)
-    test_function()
+
+    test_file_name = os.path.basename(test_file)
+    file_symbols = symbols_by_file[test_file_name]
+
+    found_pylint_message = checker_name in file_symbols
+    if not found_pylint_message:
+        print('Failed: ' + test_file)  # test doesn't say which file
+    assert found_pylint_message
```

Comment:
Rather than using `if not found_pylint_message` and printing, you can add a string to the assert statement. Something like:

```python
assert found_pylint_message, f'File {test_file} does not cause the expected pylint error.'
```